src/app/article/article-comment.component.ts

```diff
@@ -1,15 +1,17 @@
-import { Component, EventEmitter, Input, Output, OnInit, OnDestroy } from '@angular/core';
+import { Component, EventEmitter, Input, Output, OnInit, OnDestroy, ChangeDetectionStrategy, ChangeDetectorRef } from '@angular/core';
 
 import { Comment, User, UserService } from '../core';
 import { Subscription } from 'rxjs';
 
 @Component({
   selector: 'app-article-comment',
-  templateUrl: './article-comment.component.html'
+  templateUrl: './article-comment.component.html',
+  changeDetection: ChangeDetectionStrategy.OnPush
```

Comment:
can you explain the need for OnPush on every component? and subsequently the need for markForCheck?

Reply:
Its a performance improvement that helps Angular skip rendering component subtrees/components unless when one of the component's inputs changes in an immutable manner, so Angular doesn't have to do unnecessary change detection/rendering for the component.

Due to this Angular won't know that it needs to trigger change detection when an observable fires because the component's inputs didn't really change, hence the need for markForCheck()